### PR TITLE
Add some useful methods to Cuboid6

### PR DESCRIPTION
**Pull request by nikita488**

### CODE REVIEW

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public Cuboid6 expand(double d) {
     public Cuboid6 expand(Vector3 vec) {
         return expand(vec.x, vec.y, vec.z);
     }
-
+
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java:115]: Dangling space :D

File: src/main/java/codechicken/lib/vec/Cuboid6.java [hunk 2]
```diff
@@ -165,10 +165,30 @@ public boolean intersects(Cuboid6 b) {
...
+    public double volume() {
+        return (max.x - min.x + 1) * (max.y - min.y + 1) * (max.z - min.z + 1);
+    }
+    
     public Vector3 center() {
         return min.copy().add(max).multiply(0.5);
     }
-
+    
+    public double getSizeOfSide(int s) {
```
Comment (covers1624) [inline src/main/java/codechicken/lib/vec/Cuboid6.java:176]: On a second pass, getSideSize would probably be a better name for this.

**Comment by covers1624** [comment]
So whats happening with this is it ready to go after my comments are addressed?

**Comment by nikita488** [comment]
I need to think about this more. i want a method something like expandSide(EnumFacing side), so cuboid can be expanded on specific side, but i see that every side argument in this class is Integer, i want to change from int to EnumFacing, is this is right?

**Comment by covers1624** [comment]
Sure, Make sure you change the current method to use EnumFacing, rename the param name to side, and overload the method with the old int version and deprecate.

**Comment by nikita488** [comment]
Okay, i try to do this. I also want to ask.  I want something like getBoundsArray() method that will return int[] array of bounds like int[] {minX, minY, minZ, maxX, maxY, maxZ}. This is can be used to save this Cuboid in NBT (NBTTagCompound have setIntArray() and getIntArray()). But the problem is that Cuboid6 have Vector3 min and max. And Vector3 x, y, z is a Double. I don't think cast Integer to Double is a best way of doing things. Back in the days, we have Cuboid6 and CuboidCoord. CuboidCoord is a something like Cuboid6, but have BlockCoord min, max.

**Comment by covers1624** [comment]
Why not just a .toNBT() method and a constructor from an NBT tag. Instead of saving it in an array, save each min/max value separately.

**Comment by covers1624** [comment]
May aswell propogate that to Vector3aswell, then you just ask the 2 Vector3's to save to nbt and save those as a "min" and "max" tag.

**Comment by covers1624** [comment]
Infact, if thats all you need ill do that right now.

**Comment by nikita488** [comment]
toNBT() is something like myTag.setDouble("x", myVector.x), myTag.setDouble("y", myVector.y), myTag.setDouble("z", myVector.z)?

**Comment by covers1624** [comment]
Yep, Ill do that now and push it.

**Comment by nikita488** [comment]
Okay, thanks :)

**Pull request closed without merging**